File: 백준/Silver/11047. 동전 0/동전 0.py
import sys
input = sys.stdin.readline

n, m = map(int, input().split())
a = [int(input()) for _ in range(n)]

result = 0
for i in range(n - 1, -1, -1):
    div = m // a[i]
    result += div
    m -= div * a[i]

print(result)

File: 백준/Silver/11047. 동전 0/동전 0.cc
#include <iostream>
using namespace std;
using ll = long long;

const ll MAX = 11;
ll n, m, a[MAX];

int main(){
    cin >> n >> m;
    for(int i = 1;i <= n;i++) cin >> a[i];

    ll result = 0;
    for(int i = n;i >= 1;i--){
        ll div = m / a[i];
        result += div;
        m -= div * a[i];
    }

    cout << result;
    return 0;
}